engines/router/missionctl/fiberapi/traffic_splitting_strategy.go

```diff
@@ -109,47 +110,33 @@ func (s *TrafficSplittingStrategy) SelectRoute(
 	ctx context.Context,
 	req fiber.Request,
 	routes map[string]fiber.Component,
-) (fiber.Component, []fiber.Component, error) {
-	var orderedRoutes []fiber.Component
-	// array, that holds results of testing the request by each rule configured on the strategy
-	// `results[k]` – is `true` if the request satisfies `k`th rule of the strategy, and `false`
-	// otherwise
-	results := make([]bool, len(s.Rules))
-
-	for idx, rule := range s.Rules {
+) (fiber.Component, []fiber.Component, fiber.Labels, error) {
+	labels := fiber.NewLabelsMap()
+
+	for _, rule := range s.Rules {
 		res, err := rule.TestRequest(req)
 		if err != nil {
-			return nil, nil, createFiberError(err, req.Protocol())
-		}
-
-		results[idx] = res
-	}
-
-	// select primary route and fallbacks, based on the results of testing
-	// given request against traffic-splitting rules
-	for i := 0; i < len(results); i++ {
-		routeID := s.Rules[i].RouteID
-		if results[i] {
+			return nil, nil, labels, createFiberError(err, req.Protocol())
+		} else if res {
+			routeID := rule.RouteID
 			if r, exists := routes[routeID]; exists {
-				orderedRoutes = append(orderedRoutes, r)
-			} else {
-				err := errors.Newf(errors.BadConfig, `route with id "%s" doesn't exist in the router`, routeID)
-				log.WithContext(ctx).Errorf(err.Error())
-				return nil, nil, createFiberError(err, req.Protocol())
+				return r, []fiber.Component{}, labels.WithLabel(TrafficRuleLabel, r.ID()), nil
```

Comment:
not a strong opinion and related comment to the fiber PR, but i think when there are no fallbacks, we should simply return nil when there are no fallbacks

Reply:
Hmm.. I think the empty slice is closer to the signature though. No strong opinion for me either, but ultimately they both kind of behave the same. So I'll leave it as it is, since that's what we've mostly been doing in the codebase.